tooling/lsp/src/requests/completion.rs

```diff
@@ -1556,10 +1556,10 @@ fn get_field_type(typ: &Type, name: &str) -> Option<Type> {
 }
 
 fn get_array_element_type(typ: Type) -> Option<Type> {
-    match typ {
+    match typ.follow_bindings() {
```

Comment:
Instead of matching on `typ.follow_bindings()`, since we're already matching on the type we should just match on the `Type::TypeVariable` and `Type::NamedGeneric` variants explicitly. It is some more code but should be a bit faster at least.

Reply:
The third's a charm 🤞